src/gangliad/statsd.cpp

```diff
@@ -987,21 +1057,138 @@ StatsD::publishMetrics( int /* timerID */ )
 		}
 	}
 
-	CollectorList* collectors = daemonCore->getCollectorList();
-	ASSERT( collectors );
+	// Now that we have formulated the query, go out and query collector(s)
+	
+	// Handle config knob MONITOR_MULTIPLE_COLLECTORS, which is an optional knob to
+	// tell the gangliad to monitor a list of specified collectors instead of the
+	// default pool collector.  This knob should be a comma seperated list of entries,
+	// where each entry is name-of-collector/address-of-collector, where the address is
+	// same as what you would give the "-pool" parameter of condor_status.  For instance,
+	// MONITOR_MULTIPLE_COLLECTORS = ce1.wisc.edu/ce1.wisc.edu:9619, ce2.wisc.edu/ce2.wisc.edu:9619
+	bool multiple_collectors = false;
+	std::vector<std::string> collector_pool_list;
+	std::vector<std::string> collector_name_list;
```

Comment:
Ideally probably should have been a vector of tuples instead of two associative vectors (as coded), or a map (as suggested).  But there is nothing intrinsically wrong with the code as-is, and it has already been tested for weeks in production, so I do not want to change it.